### ISSUE TITLE
Investigate floating precision rounding errors in `receive_payment`

### ISSUE THREAD

**Issue by erdemyerebasmaz** [breez/breez-sdk-liquid#441]
⚠️
The issue reproducible on Rust layer as @ok300 pointed out.

I had made an error using CLI command when testing, changing the title & description of the issue.

----

Re: 
- https://github.com/breez/misty-breez/issues/142

When validating invoice amount of a receive payment request to MRH amount, there's a floating-point precision rounding error when converting MRH amount to sats & u64.

Rounding the MRH amount, `bip21_amount_btc: f64`, before it's cast to `u64` resolved the issue
```rust
 let received_bip21_amount_sat: u64 = (bip21_amount_btc * 100_000_000.0).round() as u64;
 ```

**Comment by ok300** [comment]
I could reproduce it.

```rust
let bip21_amount_btc: f64 = 0.0005981;
let bip21_amount_sat: u64 = (bip21_amount_btc * 100_000_000.0) as u64;

println!("bip21_amount_btc: {bip21_amount_btc:#?}");
println!("bip21_amount_sat: {bip21_amount_sat:#?}");
```
prints
```
bip21_amount_btc: 0.0005981
bip21_amount_sat: 59809
```
showing that precision is lost when we multiply the `f64`.

---

Once #414 is in, I can use the new `parse` method to parse the MRH BIP21 directly into a

```
dictionary LiquidAddressData {
    string address;
    Network network;
    string? asset_id;
    u64? amount_sat;
    string? label;
    string? message;
};
```

which directly gives the `amount_sat` in `u64`.

This way, we bypass entirely going through floats and back.

**Comment by ok300** [comment]
Fixed in #448
